index.html

```diff
@@ -1887,324 +1887,639 @@ <h2>
+            readonly attribute DOMString organization;
+            readonly attribute DOMString recipient;
+            readonly attribute DOMString phone;
+          };
+        </pre>
         <p>
-          The steps to <dfn>create a payment address</dfn> from user-provided
-          input are given by the following algorithm.
+          The <a>PaymentAddress</a> interface represents a <a>physical
+          address</a> on the Web.
         </p>
-        <ol data-link-for="PaymentAddress">
-          <li>Let <var>address</var> be a new instance of
-          <a>PaymentAddress</a>.
-          </li>
-          <li>Set <var>address</var>.<a>[[\addressLine]]</a> to the result of
-          splitting the user-provided address line into a <a data-cite=
-          "!WEBIDL#dfn-frozen-array-type">frozen array</a>. If none was
-          provided, set it to the empty <a data-cite=
-          "!WEBIDL#dfn-frozen-array-type">frozen array</a>.
-            <aside class="note">
-              How to split an address line is locale dependent and beyond the
-              scope of this specification.
-            </aside>
-          </li>
-          <li>Set <var>address</var>.<a>[[\country]]</a> to the user-provided
-          country as an upper case [[!ISO3166]] alpha-2 code, or to the empty
-          string if none was provided.
-          </li>
-          <li>Set the <var>address</var>.<a>[[\phone]]</a> to the user-provided
-          phone number for this shipping address, optionally formatted to
-          adhere to [[!E.164]], or to the empty string if none was provided.
-            <aside class="note" title="Privacy of phone number">
-              <p>
-                To maintain user's privacy, implementers need to be mindful
-                that a shipping address's associated phone number might be
-                different or the same from that of the end-user's. As such,
-                implementers need to take care to not provide the end user's
-                phone number to the merchant without the end user's consent.
-              </p>
-            </aside>
-          </li>
-          <li>Set <var>address</var>.<a>[[\languageCode]]</a> to a
+        <section>
+          <h2>
+            Constructor
+          </h2>
+          <p>
+            The steps to <dfn data-lt="PaymentAddress.PaymentAddress()"
+            data-no-default="">construct a <code>PymentAddress</code></dfn>
+            with <a>AddressInit</a> <var>details</var> are given by the
+            following algorithm:
+          </p>
+          <ol data-link-for="AddressInit">
+            <li>Let <var>address</var> be a new instance of
+            <a>PaymentAddress</a> with <a>[[\addressLine]]</a> set to an empty
+            <code><a data-cite=
+            "!WEBIDL#idl-sequence">sequence</a>&lt;<a data-cite=
+            "WEBIDL#idl-DOMString">DOMString</a>&gt;</code>, and all other
+            internal slots set to the empty string.
```

Comment:
Do you want to distinguish between null and empty string? Null feels a bit more natural for "unset" fields.

I have a feeling we discussed this before... apologies if so.

Reply:
We indeed have... I wanted `null` too :) But then I started playing around with integrating with `<form>` and the empty string turned out to be a better fit.